modal/cli/secret.py

```diff
@@ -91,6 +92,28 @@ def some_function():
     console.print(Syntax(example_code, "python"))
 
 
+@secret_cli.command("delete", help="Delete a named secret.")
+@synchronizer.create_blocking
+async def delete(
+    secret_name: str = Argument(help="Name of the modal.Secret to be deleted. Case sensitive"),
+    yes: bool = YES_OPTION,
+    env: Optional[str] = ENV_OPTION,
+):
+    """TODO"""
+    env = ensure_env(env)
+    secret = await _Secret.from_name(secret_name, environment_name=env).hydrate()
+    if not yes:
+        typer.confirm(
+            f"Are you sure you want to irrevocably delete the modal.Secret '{secret_name}'?",
+            default=False,
+            abort=True,
+        )
+    client = await _Client.from_env()
+
+    # TODO: replace with API on `modal.Secret` when we add it
+    await client.stub.SecretDelete(api_pb2.SecretDeleteRequest(secret_id=secret.object_id))
```

Comment:
Should we just add it to the sdk as well?

Reply:
Yeah eventually for sure, but I think we're still split on the pattern to use for "object management" methods like this. Didn't want to hold it up on that discussion.